packages/next/shared/lib/router/router.ts

```diff
@@ -1997,6 +2021,10 @@ export default class Router implements BaseRouter {
       if (parsed.pathname !== pathname) {
         pathname = parsed.pathname
         parsed.pathname = pathname
+        Object.assign(
```

Comment:
Should this assign be guarded by `if(unstable_skipClientCache)`?

Reply:
This was fixing middleware data cache key deduping and isn't related to this flag specifically so doesn't need to be flagged.